### ISSUE TITLE
Add more Round thermostats with gateway

### ISSUE THREAD

**Issue by HugoMoret** [home-assistant/home-assistant.github.io#35669]
### Feedback

I use 3 Round thermostats with 3 gateways in one building. Now only one Round thermostat is visible. How can I control all 3?

### URL

https://www.home-assistant.io/integrations/evohome/

### Version

2024.11.1

**Comment by zxdavb** [comment]
TL;DR: the **evohome** integration does not support your use case.

-------------------------------

The **evohome** product supports multiple 'controllers' like so:
 - the _user account_, that will have one or more
 - _location(s)_, which has one or more
 - _gateway(s)_, which has one or more 
 - _temperature control system(s)_ (aka TCS), which has one or more zones, +/- DHW

You appear to describe a system with 1 location (I assume), 3 gateways, and 3 TCSs.

Currently, HA's **evohome** integration supports only _a single_ evohome location (you can specify which), and only _the first_ TCS (of the first gateway) of that location.

There is a hack available to permit multiple locations, [here](https://github.com/zxdavb/evohome-async/wiki/Hass:-Multiple-Evohome-Locations), but the restriction still applies for gateways after the first gateway.

The hack exists because it not rare that people have requested support for multiple locations (i.e. a single TCS in each). 

However, I don't recall people asking for multiple TCSs in a single location.  If you want multiple gateways per location, you'll need to modify the above hack. 

This is an example of the relevant line of code in the integration, in **coordinator.py**:
```python
self.loc = self.client.locations[loc_idx]
self.loc_utc_offset = timedelta(minutes=self.loc.timeZone[UTC_OFFSET])
self.tcs = self.loc._gateways[0]._control_systems[0]  # noqa: SLF001
```

**Comment by HugoMoret** [comment]
Thank your u for the quick response!
Kind regards,

Hugo Moret

TypeApp voor Android downloaden ​

Op 9 nov 2024, 14:38, om 14:38, David Bonnes ***@***.***> schreef:

**Comment by zxdavb** [comment]
If you would like, obtain the JSON using [this method](https://github.com/zxdavb/evohome-async/wiki/Hass:-Evohome-Debug-Logs), and post it here. 

I can't promise anything, but I may be able to have another look at supporting multiple TCSs per location in the future - having your JSON to test/dev against with facilitate that.
